### PR TITLE
IBX 5283 - Describe shipping management

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | IBX 5283 | 
| Versions      | 4.5 |
| Edition       | Commerce |

Describe shipping management

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        $shipmentIdentifier = '4ac4b8a0-eed8-496d-87d9-32a960a10629';
+        $shipment = $this->shipmentService->getShipmentByIdentifier($shipmentIdentifier);
+
+        $output->writeln(sprintf('Your shipment has status %s', $shipment->getStatus()));
+
+        // Query for shipments
+        $shipmentCriterions = [
+            new ShippingMethodCriterion(1),
+            new CreatedAtCriterion('2023-03-24 15:09:16'),
+            new UpdatedAtCriterion('2023-03-25 09:00:15'),
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:63]: We have `\Ibexa\Contracts\Checkout\Shipment\Query\Criterion\ShippingMethod`, `\Ibexa\Contracts\Checkout\Shipment\Query\Criterion\CreatedAt`, `\Ibexa\Contracts\Checkout\Shipment\Query\Criterion\UpdatedAt`. If want to use them with such names they need to be imported with an alias

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        // Delete the shipping method
+        $this->shippingMethodService->deleteShippingMethod($shippingMethod);
+        $output->writeln(sprintf('Deleted shipping method with ID %d and identifier "%s".', $shippingMethod->getId(), $shippingMethod->getIdentifier()));
+
+        // Delete the shipping method translation
+        $deleteTranslationStruct = new ShippingMethodDeleteTranslationStruct();
+        $deleteTranslationStruct->shippingMethod = $shippingMethod;
+        $deleteTranslationStruct->language = 'en';
+        $
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:77]: something is missing here

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    }
+}
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:79]: No newline at end of file

File: code_samples/front/shop/shipping_management/config/packages/ibexa.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                        - pending
+                    to: ready_to_ship
+                send:
+                    from:
+                        - ready_to_ship
+                    to: shipped
+                deliver:
+                    from:
+                        - shipped
+                    to: delivered
```
Comment (mikadamczyk) [inline code_samples/front/shop/shipping_management/config/packages/ibexa.yaml:46]: No newline at end of file

File: code_samples/front/shop/shipping_management/config/packages/ibexa.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                        translation_domain: "shipment"
+                    free:
+                        name: "Free"
+                        translation_domain: "shipment"
+
+// ...
+
+framework:
+    workflows:
+        ibexa_shipment:
```
Comment (mikadamczyk) [inline code_samples/front/shop/shipping_management/config/packages/ibexa.yaml:18]: Current version https://github.com/ibexa/checkout/blob/58c6f477a5145da117cdff8201824739ca701d6e/src/bundle/Resources/config/workflow.yaml#L60

File: code_samples/front/shop/shipping_management/config/packages/ibexa.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        default: 
+            shipment:
+                workflow: ibexa_shipment
+                shipping_methods:
+                    flat_rate:
+                        name: "Flat rate"
+                        translation_domain: "shipment"
+                    free:
+                        name: "Free"
+                        translation_domain: "shipment"
```
Comment (mikadamczyk) [inline code_samples/front/shop/shipping_management/config/packages/ibexa.yaml:12]: Shipping methods are fetched from the database only

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 7]
```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+ 
+namespace App\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Ibexa\Contracts\Shipping\ShipmentServiceInterface;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:10]:
```suggestion
use Ibexa\Contracts\Checkout\ShipmentServiceInterface;
```
Currently, namespaces are not `Shipping` but `Checkout`

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+use Ibexa\Contracts\Shipping\Value\ShipmentCreateStruct;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:11]:
```suggestion
use Ibexa\Contracts\Checkout\Shipment\ShipmentCreateStruct;
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+use Ibexa\Contracts\Shipping\Value\ShipmentMetadataUpdateStruct;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:12]:
```suggestion
```
No such class in `Shipment` code base

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+use Ibexa\Contracts\Shipping\Value\ShipmentQuery;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:13]:
```suggestion
use Ibexa\Contracts\Checkout\Shipment\ShipmentQuery;
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+use Ibexa\Contracts\Shipping\Value\ShipmentUpdateStruct;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:14]:
```suggestion
use Ibexa\Contracts\Checkout\Shipment\ShipmentUpdateStruct;
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+{
+    private PermissionResolver $permissionResolver;
+
+    private UserService $userService;
+
+    private ShippingServiceInterface $shipmentService;
+
+    public function __construct(
+      PermissionResolver $permissionResolver, 
+      UserService $userService,
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:28]:
```
use \Ibexa\Contracts\Core\Repository\UserService;
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+      ShippingServiceInterface $shipmentService
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:29]:
```suggestion
      ShipmentServiceInterface $shipmentService
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+use Ibexa\Core\Repository\Permission\PermissionResolver;
+
+
+final class ShipmentCommand extends Command
+{
+    private PermissionResolver $permissionResolver;
+
+    private UserService $userService;
+
+    private ShippingServiceInterface $shipmentService;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:24]:
```suggestion
    private ShipmentServiceInterface $shipmentService;
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+
+        // Get a single shipment by identifier
+        $shipmentIdentifier = '4ac4b8a0-eed8-496d-87d9-32a960a10629';
+        $shipment = $this->shipmentService->getShipmentByIdentifier($shipmentIdentifier);
+
+        $output->writeln(sprintf('Your shipment has status %s', $shipment->getStatus()));
+
+        // Query for shipments
+        $shipmentCriterions = [
+            new ShippingMethodCriterion(1),
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:61]: `use Ibexa\Contracts\Checkout\Shipment\Query\Criterion\ShippingMethod;` import is needed and then we can use it here 

```
new ShippingMethod(1),
```
or 
`use Ibexa\Contracts\Checkout\Shipment\Query\Criterion\ShippingMethod as ShippingMethodCriterion;`
and then 
```
new ShippingMethodCriterion(1),
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+            new CreatedAtCriterion('2023-03-24 15:09:16'),
+            new UpdatedAtCriterion('2023-03-25 09:00:15'),
+        ];
+        
+        $shipmentQuery = new ShipmentQuery(new LogicalOr(...$shipmentCriterions));
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:66]:
```
use Ibexa\Contracts\Checkout\Shipment\Query\Criterion\LogicalOr;
```
import is missing

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+
+        $shipmentsList->getShipments();
+        $shipmentsList->getTotalCount();
+
+        foreach ($shipmentsList as $shipment) {
+            $output->writeln($shipment->getIdentifier() . ': ' . $shipment->getName());
+        }
+
+        // Create a new shipment
+        $shipmentCreateStruct = new ShipmentCreateStruct();
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:79]: `ShipmentCreateStruct` has 3 mandatory parameters
```
    public function __construct(
        ShippingMethodInterface $method,
        OrderInterface $order,
        Money $amount
    )
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+        $output->writeln(sprintf('Created shipment with identifier %s', $shipment->getIdentifier()));
+
+        // Update existing shipment
+        $shipmentUpdateStruct = new ShipmentUpdateStruct(
+            'prepare',
+            'newIdentifier',
+            new ArrayMap(['new' => 44])
+        );
+
+        $updatedShipment = $this->shipmentService->updateShipment($shipment, $shipmentUpdateStruct);
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:94]: `Ibexa\Contracts\Core\Collection\ArrayMap`
import is missing

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 19]
```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Command;
+
+use Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodCreateStruct;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:7]: import is not used
```suggestion
```

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+use Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodDeleteTranslationStruct;
+use Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodInterface;
+use Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodListInterface;
+use Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodQuery;
+use Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodUpdateStruct;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:12]:
```suggestion
```

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+use Ibexa\Contracts\Checkout\ShippingMethodServiceInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ShippingMethodCommand extends Command
+{
+    private PermissionResolver $permissionResolver;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:20]: `use Ibexa\Contracts\Core\Repository\PermissionResolver;`
import is missing

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+
+        // Update the shipping method
+        $updateStruct = $this->shippingMethodService->newShippingMethodUpdateStruct();
+        $updateStruct->name = 'New Standard';
+        $shippingMethod = $this->shippingMethodService->updateShippingMethod($shippingMethod, $updateStruct);
+        $output->writeln(sprintf('Updated shipping method with ID %d and new name "%s".', $shippingMethod->getId(), $shippingMethod->getName()));
+
+        // Find shipping methods
+        $query = new ShippingMethodQuery();
+        $query->orderBy('name', 'ASC');
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:65]: Method 'orderBy' not found in \Ibexa\Contracts\Checkout\Value\ShippingMethod\ShippingMethodQuery

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+        $shippingMethods = $this->shippingMethodService->findShippingMethods($query);
+        $output->writeln(sprintf('Found %d shipping methods.', $shippingMethods->count()));
+
+        // Delete the shipping method
+        $this->shippingMethodService->deleteShippingMethod($shippingMethod);
+        $output->writeln(sprintf('Deleted shipping method with ID %d and identifier "%s".', $shippingMethod->getId(), $shippingMethod->getIdentifier()));
+
+        // Delete the shipping method translation
+        $deleteTranslationStruct = new ShippingMethodDeleteTranslationStruct();
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:74]: class `ShippingMethodDeleteTranslationStruct` has mandatory constructor parameters: `$shippingMethod` and `$languageCode`

**Review by mikadamczyk — COMMENTED** [review]
Two types of shipping methods are available by default: flat rate and free shipping. The flat rate method has a fixed price. In the free shipping method, a minimum price of a cart is defined. Based on them, the user can create concrete shipping methods. If the shipping method is enabled in the admin panel and is available for a given basket, then it will be displayed on the list of available shipping methods, during checkout. For example, the currency or the value of the basket for free methods is checked.

File: docs/commerce/shipping_management/shipping_method_api.md [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+
+To update a shipping method, use the `ShippingMethodServiceInterface::deleteShippingMethod` method.
+
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/ShippingMethodCommand.php', 100, 106) =]]
+```
+
+## Delete shipping method translation
+
+To delete shipping method translation, use the `ShippingMethodServiceInterface::deleteShippingMethod` method.
```
Comment (DominikaK) [inline docs/commerce/shipping_management/shipping_method_api.md:69]:
```suggestion
To delete shipping method translation, use the `ShippingMethodServiceInterface::deleteShippingMethodTranslation` method.
```

File: docs/commerce/shipping_management/configure_shipment.md [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+---
+
+# Configure shipment
+
+When you work with your Commerce implementation, you can review and modify 
+the shipment configuration.
+
+!!! note "Permissions" 
+
+    When you modify the workflow configuration, make sure you properly set user [permissions](permission_use_cases.md#commerce) to the shipping management component.
```
Comment (DominikaK) [inline docs/commerce/shipping_management/configure_shipment.md:13]:
```suggestion
    When you modify the workflow configuration, make sure you properly set user [permissions](permission_use_cases.md#commerce) for the shipping management component.
```

File: docs/commerce/shipping_management/configure_shipment.md [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+## Shipment workflow
+
+Shipment workflow relies on a [Symfony Workflow](http://symfony.com/doc/5.4/components/workflow.html).
+Each transition represents a separate shipment step. 
+
+The default fallback workflow is `ibexa_shipment`, which is prepended at bundle level.
+
+### Default shipment workflow configuration
+
+The default shipment workflow configuration looks as follows. `ibexa_shipment` key is used by default, you can replace it with your custom workflow identifier if needed.
```
Comment (DominikaK) [inline docs/commerce/shipping_management/configure_shipment.md:24]: Like in payment, I'd avoid quoting the default workflow here, instead, later give an example under Custom workflows

File: docs/commerce/shipping_management/configure_shipment.md [hunk 27]
```diff
@@ -0,0 +1,41 @@
...
+
+You define custom workflow implementations under the `framework.workflows` key. 
+The `checkout.shipment_workflow` parameter is repository-aware.
+
+To customize your configuration, place it in a YAML file, under the `framework.workflows.<your_workflow_name>` key, and reference it with `ibexa.repositories.<your_repository>.shipment.workflow: your_workflow_name`. 
+The system can then identify which of your configured workflows handles the shipment process.
+
+## Configure shipping methods
+
+You can define the shipping methods [in the UI]([[= user_doc =]]/commerce/shipping_management/configure_shipping_method.md).
```
Comment (DominikaK) [inline docs/commerce/shipping_management/configure_shipment.md:40]:
```suggestion
You can define the shipping methods [in the UI]([[= user_doc =]]/commerce/shipping_management/configure_shipping_method/).
```

File: docs/commerce/shipping_management/shipment_api.md [hunk 28]
```diff
@@ -0,0 +1,66 @@
+---
+description: Use PHP API to manage shipments n Commerce: create, update and delete shipments.
```
Comment (DominikaK) [inline docs/commerce/shipping_management/shipment_api.md:2]:
```suggestion
description: Use PHP API to manage shipments in Commerce to create, update and delete shipments.
```

File: docs/commerce/shipping_management/shipment_api.md [hunk 29]
```diff
@@ -0,0 +1,66 @@
...
+It follows the same search query pattern as other APIs:
+
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/ShipmentCommand.php', 62, 79) =]]
+```
+
+## Create shipment
+
+To create a shipment, use the `ShipmentService::createShipment` method and provide it with 
+the `Ibexa\Contracts\Checkout\Value\ShipmentCreateStruct` object that passes two parameters, the `shippingMethod` string and `Money` object.
```
Comment (DominikaK) [inline docs/commerce/shipping_management/shipment_api.md:44]:
```suggestion
an `Ibexa\Contracts\Checkout\Value\ShipmentCreateStruct` object that takes two parameters, a `shippingMethod` string and a `Money` object.
```

File: docs/commerce/shipping_management/shipment_api.md [hunk 30]
```diff
@@ -0,0 +1,66 @@
...
+You can update the shipment after it is created. 
+You could do it to support a scenario when, for example, the shipment is processed offline and its status has to be updated in the system. 
+To update shipment information, use the `ShipmentService::updateShipment` method:
+
+``` php
+[[= include_file('code_samples/api/commerce/src/Command/ShipmentCommand.php', 92, 98) =]]
+```
+## Delete shipment
+
+To delete a shipment from the system, use the ``ShipmentService::deleteShipment` method:
```
Comment (DominikaK) [inline docs/commerce/shipping_management/shipment_api.md:61]:
```suggestion
To delete a shipment from the system, use the `ShipmentService::deleteShipment` method:
```

File: docs/commerce/shipping_management/shipping_management.md [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+---
+
+# Shipping management
+
+The shipping management component enables users to define and manage shipping methods of different types, as well as create and manage shipments, search for shipments and filter search results. 
+Depending on their role, users can also enable or disable shipping methods, change status of shipments, and cancel shipments.
+
+!!! note "Shipping method types"
+
+    Two types of shipping methods are available by default: "flat rate" and "free".
```
Comment (DominikaK) [inline docs/commerce/shipping_management/shipping_management.md:13]:
```suggestion
    Two types of shipping methods are available by default: `flat rate` and `free`.
```

File: docs/commerce/shipping_management/shipping_method_api.md [hunk 32]
```diff
@@ -0,0 +1,73 @@
...
+
+    To learn how to manage shipping with the REST API, see the [REST API reference](../../api/rest_api/rest_api_reference/rest_api_reference.html#managing-shipping).
+
+To get shipping methods and manage them, use the `Ibexa\Contracts\Checkout\ShippingMethodServiceInterface` interface.
+
+## Get shipping method
+
+### Get shipping method by identifier
+
+To access a shipping method  by using its identifier, use the `ShippingMethodServiceInterface::getShippingMethod` method.
```
Comment (DominikaK) [inline docs/commerce/shipping_management/shipping_method_api.md:18]:
```suggestion
To access a shipping method by using its identifier, use the `ShippingMethodServiceInterface::getShippingMethod` method.
```

**Review by DominikaK — CHANGES_REQUESTED** [review]

**Review by DominikaK — APPROVED** [review]

**Review by DominikaK — APPROVED** [review]

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 33]
```diff
@@ -0,0 +1,116 @@
...
+
+        // Get a single shipment by identifier
+        $identifier = '4ac4b8a0-eed8-496d-87d9-32a960a10629';
+        $shipment = $this->shipmentService->getShipmentByIdentifier($identifier);
+
+        $output->writeln(sprintf('Your shipment has status %s', $shipment->getStatus()));
+
+        // Query for shipments
+        $shipmentCriterions = [
+            new ShippingMethod($this->shippingMethodService->getShippingMethod('free'), ),
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:75]:
```suggestion
            new ShippingMethod($this->shippingMethodService->getShippingMethod('free')),
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 34]
```diff
@@ -0,0 +1,116 @@
...
+        $output->writeln(sprintf('Shipment %d has status %s', $id, $shipment->getStatus()));
+
+        // Get a single shipment by identifier
+        $identifier = '4ac4b8a0-eed8-496d-87d9-32a960a10629';
+        $shipment = $this->shipmentService->getShipmentByIdentifier($identifier);
+
+        $output->writeln(sprintf('Your shipment has status %s', $shipment->getStatus()));
+
+        // Query for shipments
+        $shipmentCriterions = [
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:74]:
```suggestion
        $shipmentCriteria = [
```

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 35]
```diff
@@ -0,0 +1,117 @@
...
+        $shippingMethodId = 1;
+        $shippingMethod = $this->shippingMethodService->getShippingMethodById($shippingMethodId);
+
+        $output->writeln(sprintf('Availability status of shipping method %d is "%s"', $shippingMethodId, $shippingMethod->isEnabled()));
+
+        // Get a single shipping method by identifier
+        $shippingMethodIdentifier = '4ac4b8a0-eed8-496d-87d9-32a960a10629';
+        $shippingMethod = $this->shippingMethodService->getShippingMethod($shippingMethodIdentifier);
+
+        $output->writeln(sprintf('Got shipping method by identifier "%s" and type "%s".', $shippingMethodIdentifier, $shippingMethod->getType()->getIdentifier()));
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:57]:
```suggestion
        $output->writeln(
            sprintf(
                'Got shipping method by identifier "%s" and type "%s".', 
                $shippingMethodIdentifier, 
                $shippingMethod->getType()->getIdentifier()
            )
        );
```

File: code_samples/api/commerce/src/Command/ShippingMethodCommand.php [hunk 36]
```diff
@@ -0,0 +1,117 @@
...
+        $shippingMethodQuery = new ShippingMethodQuery(new ShippingMethodRegion($this->regionService->getRegion('EU')));
+        $shippingMethodQuery->setLimit(10);
+
+        $shippingMethods = $this->shippingMethodService->findShippingMethods($shippingMethodQuery);
+
+        $shippingMethods->getShippingMethods();
+        $shippingMethods->getTotalCount();
+
+        foreach ($shippingMethods as $shippingMethod) {
+            $output->writeln($shippingMethod->getIdentifier() . ': ' . $shippingMethod->getName() . '- ' . $shippingMethod->getDescription());
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShippingMethodCommand.php:69]:
```suggestion
            $output->writeln(
                sprintf(
                    '%s: %s- %s',
                    $shippingMethod->getIdentifier(),
                    $shippingMethod->getName(),
                    $shippingMethod->getDescription()
                )
            );
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 37]
```diff
@@ -0,0 +1,116 @@
...
+        $this->orderService = $orderService;
+
+        parent::__construct('doc:shipment');
+    }
+
+    public function configure(): void
+    {
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:56]:
```suggestion
    protected function execute(InputInterface $input, OutputInterface $output): int
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 38]
```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Command;
+
+use Ibexa\Contracts\Checkout\Shipment\Query\Criterion\CreatedAt;
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:7]:
```suggestion
use DateTime;
use Ibexa\Contracts\Checkout\Shipment\Query\Criterion\CreatedAt;
```

File: code_samples/api/commerce/src/Command/ShipmentCommand.php [hunk 39]
```diff
@@ -0,0 +1,136 @@
...
+                'Your shipment has status %s', 
+                $shipment->getStatus()
+            )
+        );
+
+        // Query for shipments
+        $shipmentCriteria = [
+            new ShippingMethod($this->shippingMethodService->getShippingMethod('free')),
+            new CreatedAt(new \DateTime('2023-03-24 15:09:16')),
+            new UpdatedAt(new \DateTime('2023-03-25 09:00:15')),
```
Comment (mikadamczyk) [inline code_samples/api/commerce/src/Command/ShipmentCommand.php:85]: why is this class not imported?

**Review by mikadamczyk — APPROVED** [review]

File: code_samples/api/commerce/src/Command/ShipmentCommand.php (hunk 39 above)
Comment (dabrt) [reply code_samples/api/commerce/src/Command/ShipmentCommand.php:85]: It is, in line 19

**Pull request merged**